### PR TITLE
Add type hints to urllib3.contrib.ntlmpool

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: src/urllib3/contrib/ntlmpool.py [hunk 1]
```diff
@@ -64,7 +68,7 @@ def _new_conn(self):
 
         # Remove the reference to the socket, so that it can not be closed by
         # the response object (we want to keep the socket open)
-        res.fp = None
+        res.fp = None  # type: ignore
```
Comment (hramezani) [inline src/urllib3/contrib/ntlmpool.py:71]: ignore because `error: "HTTPResponse" has no attribute "fp"`

File: src/urllib3/contrib/ntlmpool.py [hunk 2]
```diff
@@ -96,20 +100,20 @@ def _new_conn(self):
                 raise Exception("Server rejected request: wrong username or password")
             raise Exception(f"Wrong server response: {res.status} {res.reason}")
 
-        res.fp = None
+        res.fp = None  # type: ignore
```
Comment (hramezani) [inline src/urllib3/contrib/ntlmpool.py:103]: ignore because `error: "HTTPResponse" has no attribute "fp"`

File: src/urllib3/contrib/ntlmpool.py [hunk 3]
```diff
@@ -96,20 +100,20 @@ def _new_conn(self):
...
-        body=None,
-        headers=None,
-        retries=3,
-        redirect=True,
-        assert_same_host=True,
-    ):
+        method: str,
+        url: str,
+        body: Optional[HTTPBody] = None,
+        headers: Optional[Dict[str, str]] = None,
```
Comment (sethmlarson) [inline src/urllib3/contrib/ntlmpool.py:112]: Mappin[str, str]?

File: src/urllib3/contrib/ntlmpool.py [hunk 4]
```diff
@@ -96,20 +100,20 @@ def _new_conn(self):
...
+        retries: int = 3,
```
Comment (sethmlarson) [inline src/urllib3/contrib/ntlmpool.py:113]: Should this be Union[Retry, bool, int] like elsewhere?

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Nice, few thoughts:

File: src/urllib3/contrib/ntlmpool.py [hunk 5]
```diff
@@ -6,10 +6,15 @@
 
 from http.client import HTTPSConnection
 from logging import getLogger
+from typing import Any, Dict, Optional, Union
 
-from ntlm import ntlm
+from ntlm import ntlm  # type:ignore
+
+from urllib3.response import BaseHTTPResponse
```
Comment (sethmlarson) [inline src/urllib3/contrib/ntlmpool.py:13]: This should be a relative import

**Review by sethmlarson — COMMENTED** [review]
Missed one thing then we're ready to merge

File: src/urllib3/contrib/ntlmpool.py (hunk 5 above)
Comment (hramezani) [reply src/urllib3/contrib/ntlmpool.py]: Done!

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**